### PR TITLE
Import metric property on Sylius

### PR DESCRIPTION
**Pull request by lruozzi9**
This PR will solve issue #82

### CODE REVIEW

File: src/DependencyInjection/WebgriffeSyliusAkeneoExtension.php [hunk 1]
```diff
@@ -99,6 +100,13 @@ final class WebgriffeSyliusAkeneoExtension extends AbstractResourceExtension imp
                 'filesystem',
             ],
         ],
+        'metric_property' => [
+            'class' => MetricPropertyValueHandler::class,
+            'arguments' => [
+                'property_accessor',
+                'webgriffe_sylius_akeneo.converter.default_unit_measurement_value_converter'
+            ],
+        ],
```
Comment (mmenozzi) [inline src/DependencyInjection/WebgriffeSyliusAkeneoExtension.php:109]: This public static array is deprecated and must not be used changed for BC (see [BC check](https://github.com/webgriffe/SyliusAkeneoPlugin/pull/83/checks?check_run_id=2554859693)).

```suggestion
```

**Review by mmenozzi — CHANGES_REQUESTED** [review]

File: src/ValueHandler/MetricPropertyValueHandler.php [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    public function supports($subject, string $attribute, array $value): bool
+    {
+        return $subject instanceof ProductVariantInterface &&
+            $attribute === $this->akeneoAttributeCode &&
+            array_key_exists('0', $value) &&
+            is_array($value[0]) &&
+            array_key_exists('data', $value[0]) &&
+            is_array($value[0]['data']) &&
+            array_key_exists('amount', $value[0]['data']) &&
+            array_key_exists('unit', $value[0]['data'])
```
Comment (mmenozzi) [inline src/ValueHandler/MetricPropertyValueHandler.php:58]: In all other value handlers there aren't these checks. We're hiding an issue this way. If an Akeneo attribute is configured with a metric attribute value handler its value must be a metric value.

**Review by mmenozzi — CHANGES_REQUESTED** [review]
@lruozzi9 add also a proper README section for this new feature. Also add that to use this new feature you need Akeneo 5.

File: README.md [hunk 3]
```diff
@@ -352,6 +353,37 @@ webgriffe_sylius_akeneo:
...
+        type: 'metric_property'
+        options:
+          akeneo_attribute_code: 'weight'
+          sylius_property_code: 'weight'
+      height:
+        type: 'metric_property'
+        options:
+          akeneo_attribute_code: 'height'
+          sylius_property_code: 'height'
+          default_unit_measurement: 'CENTIMETER'
```
Comment (mmenozzi) [inline README.md:379]: I'd not call this `default_unit_measurement` but only `unit_measurement`. I think that the word `default` could be misunderstood as the default unit measurement of that measurement unit family and that's not the case. I'd also remove this `default` work from all the parameters and variables names used for this concept in this PR.

**Review by mmenozzi — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (lruozzi9) [reply README.md:379]: At this point, I would also rename the service to UnitMeasurementValueConverter.

Comment (mmenozzi) [reply README.md:379]: I agree!

File: src/Converter/UnitMeasurementValueConverter.php [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+
+namespace Webgriffe\SyliusAkeneoPlugin\Converter;
+
+use Webgriffe\SyliusAkeneoPlugin\MeasurementFamiliesApiClientInterface;
+use Webmozart\Assert\Assert;
+
+final class UnitMeasurementValueConverter implements UnitMeasurementValueConverterInterface
+{
+    public const RECOGNIZED_OPERATORS = ['add', 'sub', 'mul', 'div'];
```
Comment (mmenozzi) [inline src/Converter/UnitMeasurementValueConverter.php:11]:
```suggestion
    private const RECOGNIZED_OPERATORS = ['add', 'sub', 'mul', 'div'];
```

File: src/Converter/UnitMeasurementValueConverter.php [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+    /**
+     * UnitMeasurementValueConverter constructor.
+     */
+    public function __construct(MeasurementFamiliesApiClientInterface $apiClient)
+    {
+        $this->apiClient = $apiClient;
+    }
+
+    public function convert(string $amount, string $unitMeasurementCode, ?string $akeneoUnitMeasurementCode): float
```
Comment (mmenozzi) [inline src/Converter/UnitMeasurementValueConverter.php:25]:
```suggestion
    public function convert(string $amount, string $sourceUnitMeasurementCode, ?string $destinationUnitMeasurementCode): float
```

File: src/Converter/UnitMeasurementValueConverter.php [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+            )
+        );
+    }
+
+    /**
+     * @param array<array{operator: string, value: string}> $operationsToDefaultUnitMeasurement
+     * @param float $value
+     * @return float
+     */
+    protected function doOperations($operationsToDefaultUnitMeasurement, float $value): float
```
Comment (mmenozzi) [inline src/Converter/UnitMeasurementValueConverter.php:79]:
```suggestion
    private function doOperations($operationsToDefaultUnitMeasurement, float $value): float
```

File: src/Converter/UnitMeasurementValueConverter.php [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+        }
+        return $value;
+    }
+
+    /**
+     * @param array<array{operator: string, value: string}> $operationsToDefaultUnitMeasurement
+     * @param float $value
+     * @return float
+     */
+    protected function doReverseOperations($operationsToDefaultUnitMeasurement, float $value): float
```
Comment (mmenozzi) [inline src/Converter/UnitMeasurementValueConverter.php:111]:
```suggestion
    private function doReverseOperations($operationsToDefaultUnitMeasurement, float $value): float
```

File: src/Converter/UnitMeasurementValueConverter.php [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+        }
+        return $value;
+    }
+
+    /**
+     * @param array{unitCode: array{code: string, labels: array<string, string>, convert_from_standard: array{operator: string, value: string}, symbol: string}} $units
+     * @param string $unitMeasurementCode
+     * @return array|array{array{operator: string, value: string}}
+     */
+    protected function getOperationsForDefaultFromUnitMeasurement(array $units, string $unitMeasurementCode): array
```
Comment (mmenozzi) [inline src/Converter/UnitMeasurementValueConverter.php:143]:
```suggestion
    private function getOperationsForDefaultFromUnitMeasurement(array $units, string $unitMeasurementCode): array
```

File: src/Converter/UnitMeasurementValueConverterInterface.php [hunk 9]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Webgriffe\SyliusAkeneoPlugin\Converter;
+
+interface UnitMeasurementValueConverterInterface
+{
+    public function convert(string $amount, string $unitMeasurementCode, ?string $akeneoUnitMeasurementCode): float;
```
Comment (mmenozzi) [inline src/Converter/UnitMeasurementValueConverterInterface.php:7]:
```suggestion
    public function convert(string $amount, string $sourceUnitMeasurementCode, ?string $destinationUnitMeasurementCode): float;
```

File: src/Resources/config/services.xml [hunk 10]
```diff
@@ -162,5 +162,9 @@
         <service id="webgriffe_sylius_akeneo.converter.value" class="Webgriffe\SyliusAkeneoPlugin\Converter\ValueConverter" >
             <argument type="service" id="translator" />
         </service>
+
+        <service id="webgriffe_sylius_akeneo.converter.unit_measurement_value_converter" class="Webgriffe\SyliusAkeneoPlugin\Converter\UnitMeasurementValueConverter">
```
Comment (mmenozzi) [inline src/Resources/config/services.xml:166]:
```suggestion
        <service id="webgriffe_sylius_akeneo.converter.unit_measurement_value" class="Webgriffe\SyliusAkeneoPlugin\Converter\UnitMeasurementValueConverter">
```

**Review by mmenozzi — CHANGES_REQUESTED** [review]

File: src/Converter/UnitMeasurementValueConverter.php (hunk 4 above)
Comment (lruozzi9) [reply src/Converter/UnitMeasurementValueConverter.php:11]: This constant is public because it is also used outside this class.

Comment (mmenozzi) [reply src/Converter/UnitMeasurementValueConverter.php:11]: It's used in the spec only. If the only reason to have it public is to use it in a spec it sounds a code smell...

**Review by mmenozzi — APPROVED** [review]

**Pull request merged**